src/nvim/edit.c

```diff
@@ -1846,47 +1848,49 @@ static void ins_ctrl_x(void)
   }
 }
 
-/*
- * Return TRUE if the 'dict' or 'tsr' option can be used.
- */
-static int has_compl_option(int dict_opt)
+/// Check that the "dict" or "tsr" option can be used.
+///
+/// @param  dict_opt  Check "dict" when true, "tsr" when false.
+static bool has_compl_option(bool dict_opt)
 {
-  if (dict_opt ? (*curbuf->b_p_dict == NUL && *p_dict == NUL
-                  && !curwin->w_p_spell
-                  )
+  if (dict_opt ? (*curbuf->b_p_dict == NUL
+                  && *p_dict == NUL && !curwin->w_p_spell)
       : (*curbuf->b_p_tsr == NUL && *p_tsr == NUL)) {
     ctrl_x_mode = 0;
     edit_submode = NULL;
     msg_attr(dict_opt ? (char_u *)_("'dictionary' option is empty")
-        : (char_u *)_("'thesaurus' option is empty"),
-        hl_attr(HLF_E));
+             : (char_u *)_("'thesaurus' option is empty"), hl_attr(HLF_E));
     if (emsg_silent == 0) {
       vim_beep(BO_COMPL);
       setcursor();
       ui_flush();
       os_delay(2000L, false);
     }
-    return FALSE;
+    return false;
   }
-  return TRUE;
+  return true;
 }
 
-/*
- * Is the character 'c' a valid key to go to or keep us in CTRL-X mode?
- * This depends on the current mode.
- */
-int vim_is_ctrl_x_key(int c)
+/// Is the character 'c' a valid key to go to or keep us in CTRL-X mode?
+/// This depends on the current mode.
+///
+/// @param  c  character to check
+///
+/// @return true if the character is a valid CTRL-X mode key-press.
+bool vim_is_ctrl_x_key(int c)
```

Comment:
PURE, or at least WARN_UNUSED. (maybe PURE would require turning the `EMSG(e_internal)` into an assert)